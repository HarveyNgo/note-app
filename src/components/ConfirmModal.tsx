import React from 'react';
import {Modal, StyleSheet, Text, View} from 'react-native';
import PrimaryButton from './PrimaryButton';
import {useTranslation} from 'react-i18next';
import {Colors} from '../constants/colors';

interface ConfirmModalProps {
  visible: boolean;
  title: string;
  message: string;
  onConfirm: () => void;
  onCancel: () => void;
}

const ConfirmModal: React.FC<ConfirmModalProps> = ({
  visible,
  title,
  message,
  onConfirm,
  onCancel,
}) => {
  const {t} = useTranslation();
  return (
    <Modal
      transparent={true}
      animationType="fade"
      visible={visible}
      onRequestClose={onCancel}>
      <View style={styles.overlay}>
        <View style={styles.modalContainer}>
          <Text style={styles.title}>{title}</Text>
          <Text style={styles.message}>{message}</Text>
          <View style={styles.buttonContainer}>
            <PrimaryButton
              style={styles.button}
              onPress={onConfirm}
              title={t('yes')}
            />

            <PrimaryButton
              style={styles.button}
              onPress={onCancel}
              title={t('cancel')}
            />
          </View>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
  },
  modalContainer: {
    width: '80%',
    backgroundColor: 'white',
    borderRadius: 10,
    padding: 20,
    alignItems: 'center',
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  message: {
    fontSize: 16,
    textAlign: 'center',
    marginBottom: 20,
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
  button: {
    flex: 1,
    marginHorizontal: 5,
    padding: 10,
    alignItems: 'center',
  },
  buttonText: {
    color: Colors.white,
    fontSize: 16,
  },
});

export default ConfirmModal;
