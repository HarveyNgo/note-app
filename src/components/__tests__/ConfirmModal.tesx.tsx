import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ConfirmModal from '../ConfirmModal';

describe('ConfirmModal', () => {
  const mockOnConfirm = jest.fn();
  const mockOnCancel = jest.fn();

  const defaultProps = {
    visible: true,
    title: 'Test Title',
    message: 'Test Message',
    onConfirm: mockOnConfirm,
    onCancel: mockOnCancel,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when visible', () => {
    const {getByText} = render(<ConfirmModal {...defaultProps} />);
    expect(getByText('Test Title')).toBeTruthy();
    expect(getByText('Test Message')).toBeTruthy();
  });

  it('does not render when not visible', () => {
    const {queryByText} = render(
      <ConfirmModal {...defaultProps} visible={false} />,
    );
    expect(queryByText('Test Title')).toBeNull();
    expect(queryByText('Test Message')).toBeNull();
  });

  it('calls onConfirm when the confirm button is pressed', () => {
    const {getByText} = render(<ConfirmModal {...defaultProps} />);
    const confirmButton = getByText('yes');
    fireEvent.press(confirmButton);
    expect(mockOnConfirm).toHaveBeenCalledTimes(1);
  });

  it('calls onCancel when the cancel button is pressed', () => {
    const {getByText} = render(<ConfirmModal {...defaultProps} />);
    const cancelButton = getByText('cancel');
    fireEvent.press(cancelButton);
    expect(mockOnCancel).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const {toJSON} = render(<ConfirmModal {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
