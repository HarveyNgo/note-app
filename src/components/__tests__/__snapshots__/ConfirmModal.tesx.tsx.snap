// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal matches the snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Test Title
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Test Message
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F94695",
              "borderRadius": 24,
              "flex": 1,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
              "paddingVertical": 8,
            }
          }
          testID="button"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "PingFang-SC-Regular",
                },
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "lineHeight": 18.2,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
          >
            yes
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F94695",
              "borderRadius": 24,
              "flex": 1,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
              "paddingVertical": 8,
            }
          }
          testID="button"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "PingFang-SC-Regular",
                },
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "lineHeight": 18.2,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
          >
            cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
